### PR TITLE
Add chain tip tracking of all peers

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Uses the peer metadata field to keep track of peer chain tip data. 

## Motivation and Context

This will record keeping and statistical analysis of peer data. 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/parser.rs [hunk 1]
```diff
@@ -1076,6 +1080,20 @@ impl Parser {
                             }
                             s.join(", ")
                         };
+                        let base_node_data_str = {
+                            let mut s = Vec::new();
+                            match peer.get_metadata(1) {
```
Comment (sdbondi) [inline applications/tari_base_node/src/parser.rs:1085]: Add an enum in tari_core to represent the keys
```rust
#[repr(u8)]
enum PeerMetadata {
   ChainMetadata = 1
}
```

```suggestion
                            match peer.get_metadata(PeerMetadata::ChainMetadata) {
```

**Review by sdbondi — COMMENTED** [review]
I don't like the hardcoded `1` for the key - I've added a suggestion.

**Review by stringhandler — APPROVED** [review]
Untested, but code looks good

File: base_layer/core/src/base_node/state_machine_service/initializer.rs [hunk 2]
```diff
@@ -95,6 +95,8 @@ where B: BlockchainBackend + 'static
             let chain_metadata_service = handles.expect_handle::<ChainMetadataHandle>();
             let node_local_interface = handles.expect_handle::<LocalNodeCommsInterface>();
             let connectivity_requester = handles.expect_handle::<ConnectivityRequester>();
+            let base_node_comms = handles.expect_handle::<CommsNode>();
+            let peer_manager = base_node_comms.peer_manager();
```
Comment (sdbondi) [inline base_layer/core/src/base_node/state_machine_service/initializer.rs:99]:
```rust
   let peer_manager =  handles.expect_handle::<Arc<PeerManager>>();
```

**Pull request merged**